src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const startYear = arguments.length < 2 ? 0 : (century - 1) * 100 + 1;
+  const endYear = arguments.length < 2 ? Infinity : startYear + 99;
+
+  const peopleFromGivenCentury = people
+    .filter(({ died, sex }) => died > startYear
+      && died < endYear && sex === 'm');
+
+  const sumOfAges = peopleFromGivenCentury
+    .reduce((a, b) => (a + (b.died - b.born)), 0);
+
```

Comment:
It takes time to find what hides after `a` and `b` variables. You can name it like `totalYears` and `person`

Reply:
good point